### PR TITLE
[MPA/OTX] Apply changes in develop to otx branch

### PR DESCRIPTION
**Pull request by goodsong81**
- OTX PR: https://github.com/openvinotoolkit/training_extensions/pull/1436

### CODE REVIEW

File: mpa/cls/inferrer.py [hunk 1]
```diff
@@ -16,7 +16,9 @@
 
 from mpa.registry import STAGES
 from mpa.cls.stage import ClsStage
-from mpa.modules.hooks.recording_forward_hooks import ActivationMapHook, FeatureVectorHook
+# from mpa.modules.hooks.recording_forward_hooks import ActivationMapHook, FeatureVectorHook
+# FIXME: Need to merge two hook modules
+from mpa.modules.hooks.auxiliary_hooks import FeatureVectorHook, ReciproCAMHook
```
Comment (goodsong81) [inline mpa/cls/inferrer.py]: @negvet @dongkwan-kim01 Which is the right one? Could you create a PR to this branch to merge XAI changes in develop vs otx branch?

File: mpa/det/inferrer.py [hunk 2]
```diff
@@ -33,9 +37,9 @@ def run(self, model_cfg, model_ckpt, data_cfg, **kwargs):
         """
         self._init_logger()
         mode = kwargs.get('mode', 'train')
-        eval = kwargs.get('eval', False)
-        dump_features = kwargs.get('dump_features', False)
-        dump_saliency_map = kwargs.get('dump_saliency_map', False)
+        eval = kwargs.pop('eval', False)
+        dump_features = kwargs.pop('dump_features', False)
+        dump_saliency_map = kwargs.pop('dump_saliency_map', False)
```
Comment (goodsong81) [inline mpa/det/inferrer.py]: @eugene123tw Specific reason?

File: mpa/modules/models/classifiers/sam_classifier.py [hunk 3]
```diff
@@ -8,7 +8,9 @@
 from mmcls.models.classifiers.base import BaseClassifier
 from mmcls.models.classifiers.image import ImageClassifier
 from mpa.modules.utils.task_adapt import map_class_names
-from mpa.modules.hooks.recording_forward_hooks import ActivationMapHook, FeatureVectorHook
+# from mpa.modules.hooks.recording_forward_hooks import ActivationMapHook, FeatureVectorHook
+# FIXME: Need to merge to hook modules
+from mpa.modules.hooks.auxiliary_hooks import FeatureVectorHook, ReciproCAMHook
```
Comment (goodsong81) [inline mpa/modules/models/classifiers/sam_classifier.py]: @negvet @dongkwan-kim01 Same here.

**Review by goodsong81 — COMMENTED** [review]
There seems no much problems except for the XAI part.

File: mpa/cls/inferrer.py (hunk 1 above)
Comment (dongkwan-kim01) [reply mpa/cls/inferrer.py]: The `recording_forward_hooks` is the updated name in the OTX side, but we can revert it to the auxiliary_hooks

**Review by negvet — APPROVED** [review]

**Review by sovrasov — APPROVED** [review]
From my side the changes are ok

**Comment by sovrasov** [comment]
@goodsong81 please take into account changes from #118 as well

File: mpa/cls/trainer.py [hunk 4]
```diff
@@ -104,10 +104,14 @@ def run(self, model_cfg, model_ckpt, data_cfg, **kwargs):
...
-
-        # updated to adapt list of datasets for the 'train'
+        train_data_cfg = Stage.get_data_cfg(cfg, "train")
+        otx_dataset = train_data_cfg.get("otx_dataset", None)
+        drop_last = False
+        dataset_len = len(otx_dataset) if otx_dataset else 0
+        # if task == h-label & dataset size is bigger than batch size
+        if train_data_cfg.get("hierarchical_info", None) and dataset_len > cfg.data.get("samples_per_gpu", 2):
+            drop_last = True
+        # updated to adapt list of dataset for the 'train'
```
Comment (goodsong81) [inline mpa/cls/trainer.py:114]: @JihwanEom Could you confirm this change?

**Pull request merged**